src/authentication/oauth2Auth.ts

```diff
@@ -246,13 +246,25 @@ export class Oauth2Auth extends AlfrescoApiClient {
                 });
             } else {
                 if (this.config.oauth2.silentLogin) {
-                    this.implicitLogin();
+                    if (!this.isInSilentLoginExceptList()) {
+                        this.implicitLogin();
+                    }
                 }
             }
         });
 
     }
 
+    isInSilentLoginExceptList(): Boolean {
+        this.config.oauth2.silentLoginExceptList.forEach( el => {
```

Comment:
instead of using el will be better described with a proper name what el is

Reply:
you can use `silentLoginExceptList.some(...)`
`foreach` loop has no `break` and it will be executed exactly  the number of elements times. Using some, the loop stops as soon your predicate returns true.